### DISCUSSION TITLE
Does the 'optimizePackageImports' option automatically convert import statements?

### DISCUSSION THREAD

**Discussion by hwookim** [vercel/next.js#58806] [Help]
### Summary

I don't think I have a good understanding of how it works yet.
If I use it like this,
```
    optimizePackageImports: ["lodash", "date-fns", "react-use"],
```

`import { isEqual } from "lodash"`, does that automatically convert the import statement to `import isEqual from "lodash/isEqual"`?
But when I use bundle-analyzer, it seems that the whole lodash is imported.
![스크린샷 2023-11-23 오후 1 26 35](https://github.com/vercel/next.js/assets/45786387/a289a3d2-aa42-4986-9504-b60e0bae75c7)

Am I misunderstanding something? Or am I using it incorrectly?

**Comment by connor-baer** [comment]
I suspect it doesn't work for lodash because it isn't an ESM package. There's another config option that should work for what you want to achieve though:

```js
  modularizeImports: {
    lodash: {
      transform: 'lodash/{{member}}',
    },
    'lodash/fp': {
      transform: 'lodash/fp/{{member}}',
    },
  },
```
